### DISCUSSION TITLE
Is asynchronous programming available for UART

### DISCUSSION THREAD

**Discussion by cvbni** [ch32-rs/ch32-hal#56] [Q&A] [answered]
how do I construct an instance of UART<'d, T, Async>?

**Comment by andelf** [comment, marked as answer]
Check out https://github.com/ch32-rs/ch32-hal/blob/main/examples/ch32x035/src/bin/uart_tx_async.rs

**Comment by cvbni** [comment]
Thank you！It work fine for me . Great!!!
